include/mqtt/buffer.hpp

```diff
@@ -35,15 +35,32 @@ class buffer : public mqtt::string_view {
           lifetime_(std::move(spa)) {
     }
 
-    buffer substr(std::size_t offset, std::size_t length = mqtt::string_view::npos) {
+    buffer substr(std::size_t offset, std::size_t length = mqtt::string_view::npos) const& {
         // range is checked in mqtt::string_view::substr.
         return buffer(mqtt::string_view::substr(offset, length), lifetime_);
     }
 
+    buffer substr(std::size_t offset, std::size_t length = mqtt::string_view::npos) && {
+        // range is checked in mqtt::string_view::substr.
+        return buffer(mqtt::string_view::substr(offset, length), std::move(lifetime_));
+    }
+
+    mqtt::string_view view() const {
```

Comment:
It doesn't automatically convert to base class?

Reply:
Ah, it should be removed.

Reply:
Fixed.